app/src/main/java/com/example/sharingang/MapActivity.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.sharingang
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.location.Location
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.databinding.DataBindingUtil
+import com.example.sharingang.databinding.ActivityMapBinding
+import com.google.android.gms.location.*
+import com.google.android.gms.tasks.CancellationTokenSource
+import com.google.android.gms.tasks.Task
+
+const val REQUEST_LOCATION:Int = 0
+class MapActivity : AppCompatActivity(){
+    private lateinit var binding: ActivityMapBinding
+    private var lastLocation: Location? = null
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+
+    // Allows the cancellation of a location request if, for example, the user exists the activity
+    private var cancellationTokenSource = CancellationTokenSource()
+
+    /* Might be useful later
+    private val locationRequest: LocationRequest = LocationRequest.create().apply {
+        interval = 5000
+        fastestInterval = 2500
+        priority = LocationRequest.PRIORITY_HIGH_ACCURACY
+    }
+    */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = DataBindingUtil.setContentView(this, R.layout.activity_map)
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+        binding.buttonGetLocalisation.setOnClickListener{
+            checkPermissionLocation()
+        }
+    }
+
+    private fun checkPermissionLocation(){
```

Comment:
Nitpick: grammatically, this should be:
```suggestion
    private fun checkLocationPermissions(){
```